signalfx-java/src/main/java/com/signalfx/signalflow/SignalFlowClient.java

```diff
@@ -0,0 +1,197 @@
+     * 
+     * @param program
+     *            - computation written in signalflow language
+     * @param start
+     *            - Optional millisecond start timestamp
+     * @param stop
+     *            - Optional millisecond stop timestamp
+     * @param resolution
+     *            - Optional desired data resolution, in milliseconds
+     * @param maxDelay
+     *            - Optional desired maximum data delay, in milliseconds
+     */
+    public void start(String program, Long start, Long stop, Integer resolution, Integer maxDelay) {
+
+        Map<String, String> params = new HashMap<String, String>();
+        addParam(params, "start", start);
+        addParam(params, "stop", stop);
+        addParam(params, "resolution", resolution);
+        addParam(params, "maxDelay", maxDelay);
+
+        this.transport.start(program, params);
+    }
+
+    /**
+     * Stop a SignalFlow computation
+     * 
+     * @param computation
+     *            - computation instance
+     * @param reason
+     *            - Optional description of why stop was called
+     */
+    public void stop(Computation computation, String reason) {
+        stop(computation.getId(), reason);
+        computation.close();
+    }
+
+    /**
+     * Stop a SignalFlow computation
+     * 
+     * @param handle
+     *            - computation id
+     * @param reason
+     *            - Optional description of why stop was called
+     */
+    public void stop(String handle, String reason) {
+
+        Map<String, String> params = new HashMap<String, String>();
+        addParam(params, "reason", reason);
+
+        this.transport.stop(handle, params);
+    }
+
+    /**
+     * Keepalive a SignalFlow computation.
+     * 
+     * @param handle
+     *            - computation id
+     */
+    public void keepalive(String handle) {
+
+        this.transport.keepalive(handle);
+    }
+
+    /**
+     * Attach to an existing SignalFlow computation.
+     * 
+     * @param handle
+     *            - computation id
+     * @param filters
+     *            - filter written in signalflow language
+     * @param resolution
+     *            - Optional desired data resolution, in milliseconds
+     * @return computation instance
+     */
+    public Computation attach(String handle, String filters, Integer resolution) {
+
+        Map<String, String> params = new HashMap<String, String>();
+        addParam(params, "filters", filters);
+        addParam(params, "resolution", resolution);
+
+        return new Computation(this.transport, handle, params, true);
+    }
+
+    /**
+     * Close this SignalFlow client.
+     */
+    public void close() {
+        this.transport.close(1001, null);
+    }
+
+    /**
+     * helper method to populate map with non-null values 
+     * 
+     * @param params - map to populate
+     * @param key
+     * @param value
+     */
+    private void addParam(Map<String, String> params, String key, Object value) {
```

Comment:
Instead of this, I'd rather see a method like this:

```java
private static Map<String, String> buildParams(Object... params) {
    Preconditions.checkArgument(params.length % 2 == 0);
    ImmutableMap.Builder<String, String> builder = new ImmutableMap.Builder<>();
    for (int i = 0 ; i < params.length ; i += 2) {
        if (params[i] != null && params[i+1] != null) {
            builder.put(params[i].toString(), params[i+1].toString());
        }
    }
    return builder.build();
}
```

Then you can just use, for example in `attach()`:

```java
public Computation attach(String handle, String filters, Integer resolution) {
    return new Computation(this.transport, handle, buildParams("filters", filters, "resolution", resolution), true);
}
```